src/utilities.cc

```diff
@@ -60,7 +60,7 @@ using std::string;
 
 _START_GOOGLE_NAMESPACE_
 
-static const char* g_program_invocation_short_name = NULL;
+static std::string g_program_invocation_short_name;
```

Comment:
How are these changes related to the file extension problem?

Reply:
> How are these changes related to the file extension problem?

this commit: 4fca1b96730814666a1823f595dba5912a272d43

Reply:
Sorry, I do not understand. You need to explain why are you changing those lines.

Reply:
g_program_invocation_short_name is a const char*, if call InitGoogleLoggingUtilities  not in main or arg0 is a local value, g_program_invocation_short_name maybe has lifetime problem.

Reply:
Please create a separate PR for this issue.

Reply:
new issue: https://github.com/google/glog/issues/633#issue-848901031
new PR: https://github.com/google/glog/pull/634#issue-607763810